src/bullsandcows/Main.java

```diff
@@ -1,7 +1,194 @@
-package bullsandcows;
+
+import java.util.*;
+//package bullsandcows;
 
 public class Main {
+
+    public static Set<Character> secretGenerator(int numbers, int possibleNumbers) {
+        /*
+        We will generate ints and then convert them to valid characters from ascii table.
+
+        ASCII codes:
+
+        0-9: 48-57 (max possibleNumbers - 10)
+        a-z: 97-122 (max possibleNumbers - 36)
+        A-Z: 65-90 (max possibleNumbers - 62) - it makes game too complicated, but can be added at any moment
+
+         */
+        int initialNumbers = numbers;
+
+        Set<Character> answer = new LinkedHashSet<>();
+
+        if (numbers > possibleNumbers | numbers < 1 | possibleNumbers < 1)
```

Comment:
It may be better to explain the meaning of this statement using an additional variable. It will make your code more readable.

Here is an example:

`boolean done = i > 30 && !hot || a < b`

and then `if (done) {`